surefire-its/src/test/java/org/apache/maven/surefire/its/JUnitPlatformEnginesIT.java

```diff
@@ -114,7 +114,8 @@ public void platform() throws VerificationException
                 + "  surefire-api-*.jar"
                 + "  surefire-logger-api-*.jar"
                 + "  common-java5-*.jar"
-                + "  junit-platform-launcher-1.3.1.jar";
+                + "  junit-platform-launcher-1.3.1.jar"
+                + "  maven-surefire-common-*.jar";
```

Comment:
The provider classpath must not have surefire-common.
This would require us ASF devs to modify this PR.
User's motivation is very valuable.

The in-plugin classpath is used in two cases:
1. to run tests without forked VM and 
2. filter tests before forking

The problem is thata subtype (`ModularClasspathConfiguration`)  `AbstractPathConfiguration.getInprocClasspath()` returns empty CP.

Reply:
@Tibor17

This would require us ASF devs to modify this PR.


we can guide @mvsoder to make the change

Reply:
The artifact `maven-surefire-common` should present on `in-plugin` CP, but it should not present on forked CP. Let's compare `AbstractSurefireMojo` with the version `2.20.1` and we should see the regression. It would be worth to control `forkCount=0/1` in the parameterized test `JUnitPlatformEnginesIT.java`. In principle the test result should be the same however the provider CP should see the difference with the present of the problematic artifact `maven-surefire-common`.

Reply:
@mvsoder 
Here is my fix in a new branch https://github.com/apache/maven-surefire/tree/SUREFIRE-1534
Can you check it out with SNAPSHOT version on your project?
We need to add integration tests and add more Mocks to `AbstractSurefireMojoJava7PlusTest`.

Reply:
Ended up with errors running the following ITs. It didn't provide a stack trace so I'm not sure what to do with them. I'll work on adding my IT as requested.:

[ERROR] Errors: 
[ERROR]   ForkModeTestNGIT>ForkModeIT.testForkCountOneReuse:137->ForkModeIT.doTest:224 » SurefireVerifier
[ERROR]   ForkModeTestNGIT>ForkModeIT.testForkModeOncePerThreadTwoThreads:107->ForkModeIT.doTest:224 » SurefireVerifier
[ERROR]   JUnit47ParallelIT.reusableThreadCountSuitesAndClassesOptimized:384 » SurefireVerifier